src/coreneuron/mechanism/mech/enginemech.cpp

```diff
@@ -123,6 +125,7 @@ int corenrn_embedded_run(int nthread,
     }
     // set coreneuron's internal variable based on neuron arguments
     corenrn_embedded = !corenrn_skip_write_model_to_disk;
+    corenrn_file_mode = file_mode;
```

Comment:
Is there any correlation between `filemode` and `corenrn_skip_write_model_to_disk` ? 
If `filemode= true`, is `corenrn_skip_write_model_to_disk` also `true` ?

Reply:
In fact, what I see from my local tests is when `coreneuron.skip_write_model_to_disk = False` there is no spike report produced in coreneuron run. Is that because [output_spikes](https://github.com/neuronsimulator/nrn/blob/2a681764b6b8c1024dcea8d78d464c1c9e8968ce/src/coreneuron/io/output_spikes.cpp#L278) and [all_spikes_return](https://github.com/neuronsimulator/nrn/blob/2a681764b6b8c1024dcea8d78d464c1c9e8968ce/src/coreneuron/io/output_spikes.cpp#L38)  ?  I am still confused what the flag `corenrn_skip_write_model_to_disk` means.

Reply:
From the sync with @pramodk , `skip_write_model_to_disk` is reserved for blue brain use cases, so I will not touch it in the `ringtest`. We will review this parameter later on from the Neurodamus repo.